src/3d/shaders/mesh/mesh.frag

```diff
@@ -185,6 +193,37 @@ vec4 colorRamp()
     return vec4(colorRampResult,1);
 }
 
+float arrows()
+{
+    ivec2 size=textureSize(arrowsGridTexture,0);
+
+    float posX=(fs_in.worldPosition.x-arrowsMinCorner.x)/arrowsSpacing;
+    float posY=(-fs_in.worldPosition.z+arrowsMinCorner.y)/arrowsSpacing;
+    int gridPosX=int(posX+0.5);
+    int gridPosY=int(posY+0.5);
+
+    if (gridPosX<0 || gridPosX>=size.x || gridPosY<0 || gridPosY>=size.y)
+        return 0.0;
+
+    vec2 textureGridPosition=vec2((gridPosX+0.5)/size.x,(gridPosY+0.5)/size.y);
+    vec4 gridValue=texture(arrowsGridTexture,textureGridPosition);
+    float scale=gridValue.x/maxVectorLength;
```

Comment:
Just thinking - wouldn't it be better to have the scaling done when preparing the texture? (one less parameter in the shader)

Reply:
Maybe, but that needs to pass the max Vector Length to the texture generator.

Reply:
done